ndcube/ndcube.py

```diff
@@ -492,71 +551,108 @@ def _generate_world_coords(self, pixel_corners, wcs, needed_axes=None, *, units)
                 array_slice[wcs.axis_correlation_matrix[idx]] = slice(None)
                 tmp_world = world[idx][tuple(array_slice)].T
                 world_coords[idx] = tmp_world
-
         if units:
             for i, (coord, unit) in enumerate(zip(world_coords, wcs.world_axis_units)):
                 world_coords[i] = coord << u.Unit(unit)
+        return world_coords
+
 
+    def _generate_world_coords(self, pixel_corners, wcs, *, needed_axes=None, units=None):
+        """
+        Private method to generate world coordinates.
+
+        Handles both dependent and independent axes.
+
+        Parameters
+        ----------
+        pixel_corners : bool
+            If one needs pixel corners, otherwise pixel centers.
+        wcs : astropy.wcs.WCS
+            The WCS.
+        needed_axes : array-like
+            The axes that are needed.
+        units : bool
+            If units are needed.
+
+        Returns
+        -------
+        array-like
+            The world coordinates.
+        """
+        # TODO: Workout why I need this twice now.
+        if isinstance(wcs, ExtraCoords):
+            wcs = wcs.wcs
+            if not wcs:
+                return ()
+        if needed_axes is None or len(needed_axes) == 0:
+            needed_axes = np.array(list(range(wcs.world_n_dim)),dtype=int)
+        axes_are_independent = []
+        pixel_axes = set()
+        for world_axis in needed_axes:
+            pix_ax = world_axis_to_pixel_axes(world_axis, wcs.axis_correlation_matrix)
+            axes_are_independent.append(len(pix_ax) == 1)
+            pixel_axes = pixel_axes.union(set(pix_ax))
+        if len(pixel_axes) == 1:
+            pixel_axes = list(pixel_axes)
```

Comment:
```suggestion
        pixel_axes = list(pixel_axes)
```
I think this conversion should happen irrespective of the number of axes.

Reply:
Updated.